controller/execute.go

```diff
@@ -448,8 +448,13 @@ func buildSource(ctx context.Context, cfg *externaldns.Config) (source.Source, e
 	// Combine multiple sources into a single, deduplicated source.
 	combinedSource := wrappers.NewDedupSource(wrappers.NewMultiSource(sources, sourceCfg.DefaultTargets, sourceCfg.ForceDefaultTargets))
 	cfg.AddSourceWrapper("dedup")
-	combinedSource = wrappers.NewNAT64Source(combinedSource, cfg.NAT64Networks)
-	cfg.AddSourceWrapper("nat64")
+	if len(cfg.NAT64Networks) > 0 {
+		combinedSource, err = wrappers.NewNAT64Source(combinedSource, cfg.NAT64Networks)
```

Comment:
Currently, other wrappers do not return errors. It would be better to define a common interface for all wrappers.

Reply:
You mean all wrapper's constructors should also return an error?